src/vs/workbench/contrib/terminal/common/scripts/shellIntegration.fish

```diff
@@ -150,13 +152,15 @@ function __vsc_update_cwd --on-event fish_prompt
 end
 
 function __vsc_update_env --on-event fish_prompt
-	__vsc_esc EnvSingleStart 1
-	for line in (env)
-		set myVar (echo $line | awk -F= '{print $1}')
-		set myVal (echo $line | awk -F= '{print $2}')
-		__vsc_esc EnvSingleEntry $myVar (__vsc_escape_value "$myVal")
+	if test $__vscode_shell_env_reporting -eq 1
```

Comment:
Can we put this on the outside so the event listener isn't set up if this isn't set?

Reply:
Yeah, I hope that makes it a bit faster [6bf76d9](https://github.com/microsoft/vscode/pull/242234/commits/6bf76d985a2e49cd81d0f7494b5a233a4e2453a9)